src/H5WebViewer.ts

```diff
@@ -44,15 +44,25 @@ export default class H5WebViewer
 
     webview.onDidReceiveMessage(async (evt: Message) => {
       if (evt.type === MessageType.Ready) {
+        const uri = webview.asWebviewUri(document.uri).toString();
+        const name = basename(document.uri.fsPath);
         const { size } = await workspace.fs.stat(document.uri);
 
         webview.postMessage({
           type: MessageType.FileInfo,
-          data: {
-            uri: webview.asWebviewUri(document.uri).toString(),
-            name: basename(document.uri.fsPath),
-            size,
-          },
+          data: { uri, name, size },
+        });
+
+        function watcher() {
+          webview.postMessage({
+            type: MessageType.FileInfo,
+            data: { uri, name, size },
```

Comment:
Sending over a new `FileInfo` object triggers a rerender (via `setState`) and a refetch of the file in the front-end.

The great news is that the UI keeps its state (e.g. selected path in explorer, selected visualization tab, etc.), so if you're visualizing a dataset and this dataset's value changes in the file, the visualization updates completely seamlessly.

Reply:
Nice !